indigo_content_api/v1/views.py

```diff
@@ -78,19 +78,9 @@ def parse_frbr_uri(self, frbr_uri):
         return frbr_uri
 
     def determine_place(self):
-        parts = self.kwargs['frbr_uri'].split('/', 2)[1].split('-', 2)
-
-        # country
-        try:
-            self.country = Country.for_code(parts[0])
-        except Country.DoesNotExist:
-            raise Http404
-
-        # locality
-        if len(parts) > 1:
-            self.locality = self.country.localities.filter(code=parts[1]).first()
-            if not self.locality:
-                raise Http404
+        parts = self.kwargs['frbr_uri'].split('/')
+        place = parts[2] if parts[1] == 'akn' else parts[1]
+        self.country, self.locality = Country.get_country_locality(place)
```

Comment:
`get_country_locality` raises a `DoesNotExist` exception for Country or Locality, which needs to be turned into an Http404